biomage/stage/stage.py

```diff
@@ -0,0 +1,297 @@
+import click
+import requests
+import json
+import yaml
+import hashlib
+import anybase32
+import base64
+import boto3
+import re
+from collections import namedtuple
+from PyInquirer import prompt
+from github import Github
+from functools import reduce
+
+SANDBOX_NAME_REGEX = re.compile(
+    r"[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*"
+)
+
+
+def recursive_get(d, *keys):
+    return reduce(lambda c, k: c.get(k, {}), keys, d)
+
+
+def download_deployments(org, repo, ref):
+    if not ref:
+        url = f"https://raw.githubusercontent.com/{org}/iac/master/releases/staging-candidates/{repo}/refs-heads-master.yaml"  # noqa: E501
+    else:
+        url = f"https://raw.githubusercontent.com/{org}/iac/master/releases/staging-candidates/{repo}/refs-pull-{ref}-merge.yaml"  # noqa: E501
+
+    s = requests.Session()
+    r = s.get(url)
+
+    Deployment = namedtuple("Deployment", ["ref", "url", "status", "text"])
+
+    return Deployment(ref=ref or "master", url=url, status=r.status_code, text=r.text)
+
+
+def compile_requirements(org, refs):
+    DEPLOYMENTS = ("ui", "api", "worker")
+
+    ref_data = {deployment: None for deployment in DEPLOYMENTS}
```

Comment:
same here: it should be `repo` and `repos`

Reply:
Fair point.